cwl/src/main/scala-2.11/wdl4s/cwl/CwlInputParsing.scala

```diff
@@ -13,18 +14,15 @@ import lenthall.validation.Checked._
 import wdl4s.wom.executable.Executable.{InputParsingFunction, ParsedInputMap}
 
 object CwlInputParsing {
-  implicit val fileD = implicitly[Decoder[File]]
-  implicit val stringD = implicitly[Decoder[String]]
-  implicit val intD = implicitly[Decoder[Int]]
-  implicit val longD = implicitly[Decoder[Long]]
-  implicit val floatD = implicitly[Decoder[Float]]
-  implicit val doubleD = implicitly[Decoder[Double]]
-  implicit val booleanD = implicitly[Decoder[Boolean]]
-  implicit val inputArray = implicitly[Decoder[Array[MyriadInputValuePrimitives]]]
 
-  // Decodes the input file, and build the ParsedInputMap
-  private [cwl] lazy val inputCoercionFunction: InputParsingFunction = inputFile => {
-    yaml.parser.parse(inputFile).flatMap(_.as[Map[String, MyriadInputValue]]) match {
+  implicit val x = implicitly[Decoder[File]]
+
+  private [cwl] lazy val inputCoercionFunction: InputParsingFunction =
+    inputFile => {
+    yaml.
+      parser.
+      parse(inputFile).
+      flatMap(json => decode[Map[String, MyriadInputValue]](io.circe.Printer.noSpaces.pretty(json))) match {
```

Comment:
do we really have to parse the yaml, write it back to json, and reparse it again ? :/

Reply:
No this is the naive way I was doing originally and in fact your way is correct.  changing this